plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixClient.java

```diff
@@ -207,6 +215,27 @@ public PhoenixClient(PhoenixConfig config, ConnectionFactory connectionFactory,
                 identifierMapping);
         this.configuration = new Configuration(false);
         getConnectionProperties(config).forEach((k, v) -> configuration.set((String) k, (String) v));
+        JdbcTypeHandle bigintTypeHandle = new JdbcTypeHandle(Types.BIGINT, Optional.of("bigint"), Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty());
+        this.aggregateFunctionRewriter = new AggregateFunctionRewriter<>(
+                this::quoted,
+                ImmutableSet.<AggregateFunctionRule<JdbcExpression>>builder()
+                        .add(new ImplementCountAll(bigintTypeHandle))
+                        .add(new ImplementCount(bigintTypeHandle))
+                        .build());
+    }
+
+    @Override
+    public Optional<JdbcExpression> implementAggregation(ConnectorSession session, AggregateFunction aggregate, Map<String, ColumnHandle> assignments)
+    {
+        // TODO support complex ConnectorExpressions
+        return aggregateFunctionRewriter.rewrite(session, aggregate, assignments);
```

Comment:
Is it enough to rewrite the function here?
The aggregate will be executed per partition (either region or guidepost in Phoenix) and then would need to be post-aggregated.
(Apologies if I'm missing something obvious)

Reply:
Indeed. We need to add a final aggregation in `ConnectorMetadata#applyAggregation` to fix this.

Reply:
Sorry messed up the PR. Created a new one - https://github.com/trinodb/trino/pull/10964/files